### PR TITLE
Task-59006: when scanning a wrong QR code, an error message display

### PR DESCRIPTION
**Pull request by wbenabdessalem**
Fix: Add check about the scanned qr code

### CODE REVIEW

File: app/src/main/java/org/exoplatform/activity/ScannerActivity.java [hunk 1]
```diff
@@ -96,32 +97,37 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
+                ToneGenerator tg = new ToneGenerator(AudioManager.STREAM_NOTIFICATION, 100);
+                tg.startTone(ToneGenerator.TONE_PROP_BEEP);
+                final String detected_url = result.getText()  + "&source=qrcode";
+                String urlDomain = null;
+                try {
+                    System.out.println("detected_url ===========> " + detected_url);
+                    urlDomain = new URL(detected_url).getHost();
+                    detectedURL.setText(urlDomain);
+                } catch (MalformedURLException e) {
+                    e.printStackTrace();
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/activity/ScannerActivity.java]: We should not print exceptions, please use Logger instead.

File: app/src/main/java/org/exoplatform/activity/ScannerActivity.java [hunk 2]
```diff
@@ -96,32 +97,37 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
+                handler.postDelayed(new Runnable() {
+                    public void run() {
+                        Intent mIntent = new Intent();
+                        mIntent.putExtra("keyQRCode", detected_url);
+                        setResult(RESULT_OK, mIntent);
+                        finish();
+                    }
+                }, 3000);
+            }else{
+                detectedURL.setText("Not valid URL");
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/activity/ScannerActivity.java]:
```suggestion
                detectedURL.setText("Invalid URL");

```

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/exoplatform/activity/ScannerActivity.java (hunk 2 above)
Comment (ahamdi) [reply app/src/main/java/org/exoplatform/activity/ScannerActivity.java]: We need to internationalie this text.

Comment (wbenabdessalem) [reply app/src/main/java/org/exoplatform/activity/ScannerActivity.java]: Done

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -54,6 +55,7 @@
     <string name="Word.Done">Done</string>
     <string name="Word.Update">Update</string>
 
+    "Invalid URL"
```
Comment (ahamdi) [inline app/src/main/res/values/strings.xml]: This is an invalid XML structure, would it work for strings internationalization ?

Comment (wbenabdessalem) [reply app/src/main/res/values/strings.xml]: comment to ignore and should be deleted

**Review by ahamdi — APPROVED** [review]

**Pull request merged**